cinema/views.py

```diff
@@ -124,15 +161,23 @@
     )
     serializer_class = OrderSerializer
     pagination_class = OrderPagination
+    authentication_classes = (TokenAuthentication,)
+
+    def get_permissions(self) -> list:
+        if self.action in ("list", "create"):
+            permission_classes = [IsAdminOrIfAuthenticatedReadOnly]
+        else:
+            raise NotFound("Method not allowed")
```

Comment:
In the OrderViewSet, replace `raise NotFound("Method not allowed")` with `MethodNotAllowed` to provide a more accurate error response for unsupported actions.